Yosemite/Yosemite/Stores/ProductStore.swift

```diff
@@ -69,6 +73,54 @@ private extension ProductStore {
         }
     }
 
+    /// Synchronizes the Products found in a specified Order.
+    ///
+    func synchronizeProductsFor(_ order: Order, onCompletion: @escaping (Error?) -> Void) {
+        let itemIDs = order.items.map { $0.itemID }
+        let productIDs = itemIDs.uniqued()  // removes duplicate product IDs
+
+        let storage = storageManager.viewStorage
+        var missingIDs = [Int]()
+        for productID in productIDs {
+            let storageProduct = storage.loadProduct(siteID: order.siteID, productID: productID)
+            if storageProduct == nil {
+                missingIDs.append(productID)
+            }
+        }
+
+        let remote = ProductsRemote(network: network)
+        remote.loadProducts(for: order.siteID, by: missingIDs) { [weak self] (products, error) in
+            guard let products = products else {
+                onCompletion(error)
+                return
+            }
+
+            self?.upsertStoredProductsInBackground(readOnlyProducts: products, onCompletion: {
+                onCompletion(nil)
+            })
+        }
+    }
+
+    /// Retrieves multiple products with a given siteID + productIDs.
+    /// - Note: This is NOT a wrapper for retrieving a single product.
```

Comment:
A genuine question here, what do you mean by "this is not a wrapper for retrieving a single product"?

Reply:
Many of our remote request methods do things like this:

```
func retrieveOrder(orderID: Int) {
    let remote = OrdersRemote(network: network)
    remote.loadOrder(by: orderID)
}
```

and then the plural form is just a wrapper, so when requesting multiple orders, it looks like this:

```
func retrieveOrders(orderIDs: [Int]){
    // this func calls the single "rerieverOrder(by: OrderID)"
}
```

why it's important to note that the retrieve products is **not** a wrapper is because 1. It's only making a single API request instead of multiple individual ones. 2. Since it's a single API request that does not make multiple individual requests, we have to be mindful of Jetpack's 5 second timeout and not overload the array with too many products that are requested.